### PR TITLE
Added XY Strategy Agent for Negative APY

### PR DESCRIPTION
**Pull request by virendrapatidar**
- Added XY Strategy Agent for Negative APY
PS : As of now no XY Strategy is deployed on mainnet or avalanche.

### CODE REVIEW

File: Vesper-Agents-Suite/xy-loss-agent/src/agent.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    vesperFetcher.getStrategiesV3(blockNumber)
+  ]);
+  V3 = Array.from(new Set<string>(V3));
+  const valueV3promise = V3.map(async (strat: string) => {
+    const sContract = new web3.eth.Contract(
+      Strategy_ABI,
+      strat,
+    );
+
+    const name: string = await sContract.methods.NAME().call({}, blockNumber);
```
Comment (virendrapatidar) [inline Vesper-Agents-Suite/xy-loss-agent/src/agent.ts]: @patidarmanoj10 
Main code for review from Line 38 to 53

File: Vesper-Agents-Suite/xy-loss-agent/src/agent.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    const borrowCTokenAddress: string = await sContract.methods.borrowCToken().call({}, blockNumber);
+    const borrowCToken = new web3.eth.Contract(
+      Borrow_Token_ABI,
+      borrowCTokenAddress,
+    );
+    const debtOfBorrowedToken = new BigNumber(await borrowCToken.methods.borrowBalanceStored(strat).call({}, blockNumber));
+    const poolContract = new web3.eth.Contract(poolAbi as any, pool);
+    const pricePerShare = new BigNumber(await poolContract.methods.pricePerShare().call({}, blockNumber));
+    const strategyBalance = new BigNumber(await poolContract.methods.balanceOf(strat).call({}, blockNumber));
+    const borrowedTokenAsset = pricePerShare.multipliedBy(strategyBalance).dividedBy(new BigNumber('1e18'))
```
Comment (jcvernaleo) [inline Vesper-Agents-Suite/xy-loss-agent/src/agent.ts]: Do we know that the borrowedToken will always be an 18 decimal token?

**Review by jcvernaleo — CHANGES_REQUESTED** [review]
One question but looks good otherwise.

File: Vesper-Agents-Suite/xy-loss-agent/src/agent.ts (hunk 2 above)
Comment (virendrapatidar) [reply Vesper-Agents-Suite/xy-loss-agent/src/agent.ts]: Yes, all vesper pools are 18 decimals.
This needs some other updates related to the pool. 
Need to consider earning from both X and y Pool.

**Comment by jcvernaleo** [comment]
@virendrapatidar are you still hard coding a decimal in it?

**Comment by jcvernaleo** [comment]
@virendrapatidar are you still hard coding a decimal in it?

File: Vesper-Agents-Suite/xy-loss-agent/src/agent.ts (hunk 2 above)
Comment (virendrapatidar) [reply Vesper-Agents-Suite/xy-loss-agent/src/agent.ts]: All pools are 18 decimals so we are good.
Detecting loss on borrow is sufficient . No need to include apy of x.

**Review by jcvernaleo — APPROVED** [review]
OK

**Comment by jcvernaleo** [comment]
Discussed with @virendrapatidar elsewhere and despite my initial question, all good with the hardcoded 18 decimals.

**Pull request merged**